tests/phpunit/Fixtures/Process/DummyProcessBearer.php

```diff
@@ -0,0 +1,37 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Infection\Tests\Fixtures\Process;
+
+use Infection\Process\Runner\Parallel\ProcessBearer;
+use PHPUnit\Framework\Assert;
+use Symfony\Component\Process\Process;
+use function Safe\sprintf;
+
+final class DummyProcessBearer implements ProcessBearer
+{
+    private $process;
+    private $expectTimeOut;
+
+    public function __construct(Process $process, bool $expectTimeOut)
+    {
+        $this->process = $process;
+        $this->expectTimeOut = $expectTimeOut;
+    }
+
+    public function getProcess(): Process
+    {
+        return $this->process;
+    }
+
+    public function markAsTimedOut(): void
+    {
+        if (!$this->expectTimeOut) {
+            Assert::fail(sprintf(
+                'Did not expect "%s()" to be called',
+                __FUNCTION__
+            ));
+        }
+    }
+}
```

Comment:
Easier than a mock?

Reply:
I think so. In general I don't lick mocks that much, if they are appropriate why not, but if you can easily do without I prefer without. It has several advantages:

- Rots less easily
- Easier to refactor
- Less magic (so more accessible)
- Faster (mocking is slow)

I also got the impression that [jakzal](https://github.com/jakzal), [frankdejonge](https://github.com/frankdejonge) & [sebastian](https://github.com/sebastianbergmann) recommend the same. I would not say as far as blindly do without mocks at all, but I'm considering hunting down some cases where I feel we should not need mock

Reply:
I'm not a big fan of mocks tbh, so it's just surprising to see a different approach for once.